### PR TITLE
:hammer: Refactor python & markdown editors

### LINKED ISSUE #162: Pressing "Left" key when editing a markdown/title shifts view
Pressing the left arrow key when editing a markdown block will move the view to the left.
Same thing when editing the title of a block

Editing should override view navigation using arrow keys

### PR DESCRIPTION
**Pull request by FabienRoger**
Fixes #162

### CODE REVIEW

File: pyflow/core/mdeditor.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    def mousePressEvent(self, event: QMouseEvent) -> None:
+        """PythonEditor reaction to PyQt mousePressEvent events."""
+        if event.buttons() & Qt.MouseButton.LeftButton:
+            self.mode = "EDITING"
+        return super().mousePressEvent(event)
+
+    def focusOutEvent(self, event: QFocusEvent):
+        """PythonEditor reaction to PyQt focusOut events."""
+        self.mode = "NOOP"
+        return super().focusOutEvent(event)
```
Comment (MathisFederico) [inline pyflow/core/mdeditor.py]: This should be in pyflow.blocks not pyflow.core (and I guess pyeditor should also be at pyflow.blocks.codeblock.pyeditor as it is only used in codeblock)

File: pyflow/blocks/widgets/blocktitle.py [hunk 2]
```diff
@@ -22,6 +22,7 @@ class Title(QLineEdit, Serializable):
     def __init__(
         self,
         text: str,
+        block_item: QGraphicsItem,
```
Comment (MathisFederico) [inline pyflow/blocks/widgets/blocktitle.py]:
```suggestion
        parent: QGraphicsItem,
```

**Review by MathisFederico — CHANGES_REQUESTED** [review]
A few changes I see

**Comment by MathisFederico** [comment]
My previous comment on renaming block_item is not resolved

**Comment by FabienRoger** [comment]
I changed the name to parent_block because parent is already a field of the QLineEdit class.

(The issue Codacy reports is not a real issue. It doesn't understand that the view imported when typing is on is only used for List["View"], for which black (or something else in my python environment?) requires an import.)

**Review by MathisFederico — APPROVED** [review]
A nice fix and refactor !

**Pull request merged**